flink-container/src/main/java/org/apache/flink/container/entrypoint/StandaloneJobClusterEntryPoint.java

```diff
@@ -116,6 +107,20 @@ public static void main(String[] args) {
 		ClusterEntrypoint.runClusterEntrypoint(entrypoint);
 	}
 
+	public static StandaloneJobClusterConfiguration getEntrypointClusterConfiguration(String[] args) {
+		final CommandLineParser<StandaloneJobClusterConfiguration> commandLineParser = new CommandLineParser<>(new StandaloneJobClusterConfigurationParserFactory());
```

Comment:
It is true that we only need `CONFIG_DIR_OPTION` and `DYNAMIC_PROPERTY_OPTION` for memory configuration. The problem is that, different `ParserFactories` have different set of command line options, and a parser will stop parsing once it sees a un-recognized token, dropping the rest of the arguments.

E.g., `JOB_ID_OPTION` is added in `StandaloneJobClusterConfigurationParserFactory` but not `EntrypointClusterConfigurationParserFactory`. If the `StandaloneJobClusterEntryPoint` is launched with a `-jid` followed by a `-D property=value`, a parser created by `EntrypointClusterConfigurationParserFactory` will stop parsing when it sees `-jid` because it doesn't recognize it as an option, thus the dynamic property will not be parsed.

It is true that we have not passed the command line options to the bash utils. But I think we have agreed to do so in FLINK-16874.

I think moving `BashJavaUtils` is a good idea. Thanks for the suggestion, I'll try it out.

Reply:
Alright, thanks for the explanation. True, stopping of parsing because of an unknown option is a problem.